### DISCUSSION TITLE
Eglot in JSON files

### DISCUSSION THREAD

**Discussion by andrewbanchich** [joaotavora/eglot#1357] [Q&A] [answered]
When I open a JSON file like `package.json` in a TypeScript project, Eglot does not turn on. That makes sense to me. But when I open a `foo.ts` file, Eglot turns on, and then when I open `package.json`, Eglot is running in that buffer and trying to analyze it like TypeScript. Flymake diagnostics are displayed underlining all the `:` and saying `; expected`.

Is there something I need to configure to tell Eglot to not run on JSON / other config files?

**Comment by joaotavora** [comment, marked as answer]
I believe this is exactly what happens here: https://debbugs.gnu.org/cgi/bugreport.cgi?bug=67463

Long story short:

1. should be fixed  on latest master.  
2. there are workarounds

**Reply by andrewbanchich** [reply]
Awesome, thank you!

**Reply by andrewbanchich** [reply]
I'm still experiencing this, and have `straight` set up to auto update packages every time I open Emacs. `package-show-package-list` shows

```
eglot                          1.17           available    gnu      The Emacs Client for LSP servers`
...
eglot                          1.12.29        built-in              The Emacs Client for LSP servers
```

**Reply by joaotavora** [reply]
I know that the auto-update behaviour of Eglot in Emacs 29 has the misfortune that it doesn't update like it did in Emacs 28. I tried to fix this, but ultimately was prevented from doing so by higher powers.  You could raise a bug report with the Emacs maintainers, or just use use `M-x eglot-upgrade-eglot`, which is a command I did manage to put it to make this easier.

You can also find out in the Emacs 29 NEWS about some variables you can tweak to make this behave sanely.  Or you can just press `i` on the 1.17 version, then `x`. I think.

I don't know about straight though . Maybe it works, maybe it doesn't.  I don't use or support it.

**Reply by andrewbanchich** [reply]
I think even though it says `available` instead of `installed`, it will still use `1.17` - all packages installed by `straight` say `available`, and I only have one package which says `installed` (`compat`). I wanted to test this out, so I just removed the part of my config which was using `straight` to install it altogether, then I reopened Emacs, did `eglot-upgrade-eglot`, it said upgrading to 1.17, and checked `package-show-package-list` which now says:

```
eglot                          1.17           installed             The Emacs Client for LSP servers
...
eglot                          1.12.29        built-in              The Emacs Client for LSP servers
```

Opened a `.ts` file, started Eglot manually, then switched to `package.json` and still see the same errors.

**Reply by joaotavora** [reply]
> all packages installed by straight say available

That's really bizarre, but suit yourself.

> d still see the same errors.

Ah sorry, those errors.  When I mentioned fixed on master I meant Emacs master.  Meaning you have to compile your own or get a recent snapshot build for your distro.  Read the bug report I linked to.  There are also workarounds there.

**Reply by andrewbanchich** [reply]
Ah, I see. Okay, thanks!
